test/unit-test/cellular_common_api_utest.c

```diff
@@ -0,0 +1,921 @@
+
+    TEST_ASSERT_EQUAL( CELLULAR_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case for Cellular_CommonATCommandRaw.
+ */
+void test_Cellular_CommonATCommandRaw_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+    char pPrefix[] = "AtTSest";
+    char pData[] = "Test Data";
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_AtcmdRequestWithCallback_IgnoreAndReturn( CELLULAR_PKT_STATUS_OK );
+    _Cellular_TranslatePktStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonATCommandRaw( &context, pPrefix, pData, 0, NULL, NULL, 0 );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );
+}
+
+/**
+ * @brief Test that null cellular handler case for Cellular_CommonCreateSocket.
+ */
+void test_Cellular_CommonCreateSocket_Null_CellularHandler( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_INVALID_HANDLE );
+
+    cellularStatus = Cellular_CommonCreateSocket( NULL, 0, 0, 0, 0, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INVALID_HANDLE, cellularStatus );
+}
+
+/**
+ * @brief Test that null socket handler case for Cellular_CommonCreateSocket.
+ */
+void test_Cellular_CommonCreateSocket_Null_SocketHandler( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonCreateSocket( &context, 0, 0, 0, 0, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test that invalid context id case for Cellular_CommonCreateSocket.
+ */
+void test_Cellular_CommonCreateSocket_Invalid_ContextId( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+    struct CellularSocketContext socketHandle;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_IsValidPdn_IgnoreAndReturn( CELLULAR_BAD_PARAMETER );
+
+    cellularStatus = Cellular_CommonCreateSocket( &context, 0, 0, 0, 0, ( CellularSocketHandle_t * ) &socketHandle );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INVALID_HANDLE, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case for Cellular_CommonCreateSocket.
+ */
+void test_Cellular_CommonCreateSocket_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+    struct CellularSocketContext socketHandle;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_IsValidPdn_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_CreateSocketData_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonCreateSocket( &context, 0, 0, 0, 0, ( CellularSocketHandle_t * ) &socketHandle );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );

```

Comment:
What does this test do - it checks that the above 3 functions get called. It does not verify whether these functions do the expected job?